internal/k8s/reconciler/route.go

```diff
@@ -354,6 +377,36 @@ func (r *K8sRoute) Validate(ctx context.Context) error {
 				r.references.Add(routeRule, *reference)
 			}
 		}
+	case *gw.TCPRoute:
+		if len(route.Spec.Rules) != 1 {
+			err := service.NewResolutionError("a single tcp rule is required")
+			r.resolutionErrors.Add(err)
+			return nil
+		}
+
+		rule := route.Spec.Rules[0]
+
+		if len(rule.BackendRefs) != 1 {
```

Comment:
This is a constraint imposed by Consul right? We should comment that at a minimum. Idk if we've already started a list of spec deviations but this should probably be documented somewhere more visible (eventually).

Reply:
Yep, this is a Consul constraint as you can only specify a single service in an `ingress_gateway` config entry for TCP services and service splitters don't allow for TCP -- so we basically can support 1 TCPRoute/gateway, with 1 routing rule and 1 backend ref -- it's unfortunate, but our current constraint. 👍 on the documentation comment.